deployment/modules/gcp/cloudbuild/main.tf

```diff
@@ -119,25 +107,67 @@ resource "google_cloudbuild_trigger" "build_trigger" {
       wait_for = ["docker_build_conformance_gcp"]
     }
 
-    ## Deploy container image to Cloud Run.
-    ## TODO: Remove this as the `terragrunt apply` will bring up the Cloud Run.
+    ## Apply the deployment/live/gcp/ci terragrunt config.
+    ## This will bring up the conformance infrastructure, including a service
+    ## running the conformance server docker image built above.
     step {
-      id         = "cloud_run_deploy"
-      name       = "gcr.io/google.com/cloudsdktool/cloud-sdk"
-      entrypoint = "gcloud"
-      args = [
-        "run",
-        "deploy",
-        "${var.docker_env}-static-ct",
-        "--image",
-        "${local.conformance_gcp_docker_image}:$SHORT_SHA",
-        "--region",
-        var.location
+      id     = "terraform_apply_conformance_ci"
+      name   = "alpine/terragrunt"
+      script = <<EOT
+        terragrunt --terragrunt-non-interactive --terragrunt-no-color apply -auto-approve -no-color 2>&1
+        terragrunt --terragrunt-no-color output --raw conformance_url -no-color > /workspace/conformance_url
+      EOT
+      dir    = "deployment/live/gcp/ci"
+      env = [
+        "GOOGLE_PROJECT=${var.project_id}",
+        "TF_IN_AUTOMATION=1",
+        "TF_INPUT=false",
+        "TF_VAR_project_id=${var.project_id}"
       ]
       wait_for = ["docker_push_conformance_gcp"]
     }
 
-    ## TODO: Apply the terragrunt configuration to create the CI environment.
+    ## Since the conformance infrastructure is not publicly accessible, we need to use 
+    ## bearer tokens for the test to access them.
+    ## This step creates those, and stores them for later use.
+    step {
+      id       = "bearer_token"
+      name     = "gcr.io/cloud-builders/gcloud"
+      script   = <<EOT
+      curl -H "Metadata-Flavor: Google" "http://metadata.google.internal/computeMetadata/v1/instance/service-accounts/${local.cloudbuild_service_account}/identity?audience=$(cat /workspace/conformance_url)" > /workspace/cb_identity
+      EOT
+      wait_for = ["terraform_apply_conformance_ci"]
+    }
+
+    ## Test against the conformance server.
+    ## TODO: Replace this with CT Hammer when it is ready.
+    step {
+      id       = "curl_test"
+      name     = "curlimages/curl"
+      script   = <<EOT
+        curl -iX POST $(cat /workspace/conformance_url)/ci-${var.project_id}/ct/v1/add-pre-chain -H "Authorization: Bearer $(cat /workspace/cb_identity)"
```

Comment:
Let's generate a chain certificate with openssl to start with and/or store a chain in a test directory? there might already be one over there, we just need to be mindful of notBefore/notAfter.

Reply:
Done but I think it's not worth to invest time on this as we're going to replace this with CT Hammer soon.